packages/wallet-ts/src/utils/wallets/welldone/WelldoneWallet.ts

```diff
@@ -0,0 +1,201 @@
+    const welldone = await this.getWelldoneWallet()
+    try {
+        const {hash} = await welldone.request(this.chainName, {
+          method: 'dapp:sendSignedTransaction',
+          params: [
+            `0x${Buffer.from(
+              CosmosTxV1Beta1Tx.TxRaw.encode(txRaw).finish(),
+            ).toString('hex')}`,
+          ],
+        })
+        return hash
+    } catch (e) {
+      throw new CosmosWalletException(new Error((e as any).message), {
+        context: 'WELLDONE',
+        contextModule: 'bradcast-tx',
+      })
+    }
+  }
+
+  public async waitTxBroadcasted(txHash: string): Promise<TxResponse> {
+    const endpoints = await this.getChainEndpoints()
+
+    return new TxRestApi(endpoints.rest).fetchTxPoll(txHash)
+  }
+
+  private changePubKeyType (authInfoBytes: Uint8Array) {
+    const authInfo = CosmosTxV1Beta1Tx.AuthInfo.decode(authInfoBytes)
+    authInfo.signerInfos.forEach((signerInfo) => {
+      if (signerInfo.publicKey) {
+        signerInfo.publicKey.typeUrl = '/cosmos.crypto.secp256k1.PubKey'
+      }
+    })
+
+    return CosmosTxV1Beta1Tx.AuthInfo.encode(authInfo).finish()
+  }
+
+  public async signTransaction(
+    signDoc: Omit<CosmosTxV1Beta1Tx.SignDoc, 'accountNumber'> & {
+      accountNumber: Long
+    },
+  ): Promise<DirectSignResponse> {
+    const authInfoBytes = this.changePubKeyType(signDoc.authInfoBytes)
+    signDoc.authInfoBytes = authInfoBytes
+
+    const welldone = await this.getWelldoneWallet()
+    try {
+      const signBytes = makeSignBytes(signDoc)
+      const response = await welldone.request(this.chainName, {
+        method: 'dapp:signTransaction',
+        params: [`0x${Buffer.from(signBytes).toString('hex')}`],
+      })
+      return {
+        signed: signDoc,
+        signature: {
+          pub_key: encodeSecp256k1Pubkey(
+            Buffer.from(response[0].publicKey.replace('0x', ''), 'hex'),
+          ),
+          signature: Buffer.from(
+            response[0].signature.replace('0x', ''),
+            'hex',
+          ).toString('base64'),
+        },
+      }
+    } catch (e) {
+      throw new CosmosWalletException(new Error((e as any).message), {
+        context: 'WELLDONE',
+        contextModule: 'sign-tx',
+      })
+    }
+  }
+
+  public async getKey() {
+    const account = await this.getAccounts()
+    return account.pubKey
+  }
+
+  public async getChainEndpoints(): Promise<{ rpc: string; rest: string }> {
+    const { chainId } = this
+
+    try {
+      return getEndpointsFromChainId(chainId)
+    } catch (e: unknown) {
+      throw new CosmosWalletException(new Error((e as any).message), {
+        context: 'WELLDONE',
+        contextModule: 'get-chain-endpoints',
+      })
+    }
+  }
+
+  // WELLDONE Wallet only supports injective, cosmos, juno netoworks.
+  public async checkChainIdSupport() {
+    const { chainId } = this
+    if (chainId.includes('cosmoshub') || chainId.includes('juno')) {
```

Comment:
return `['injective', 'cosmoshub', 'juno'].some(cosmosChainId => chainId.includes(cosmosChainId))` is much better